client/src/pages/Search/AddressSearch.js

```diff
@@ -2,8 +2,27 @@ import { IconButton, InputAdornment, TextField } from "@mui/material";
 import { usePlacesWidget } from "react-google-autocomplete";
 import { HUNTER_COLLEGE_ADDRESS } from "./constants";
 import ClearIcon from "@mui/icons-material/Clear";
+import getUserCoordinates from "../../utils/getUserCoordinates";
+import axios from "axios";
+import MyLocationIcon from "@mui/icons-material/MyLocation";
 
 export default function AddressSearch({ updateFields }) {
+  async function userLocation({updateFields}) {
+    try {
+      const coordinates = await getUserCoordinates();
+      const latitude = coordinates.coordinates.latitude;
+      const longitude = coordinates.coordinates.longitude;
+      updateFields({longitude, latitude});
+      const reversegeolocate = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${latitude},${longitude}&key=${process.env.REACT_APP_GOOGLE_API_KEY}`;
+      axios.get(reversegeolocate).then(function (response) {
+        const address = response.data.results[0].formatted_address;
+        googlePlacesRef.current.value = address;
```

Comment:
how are you accessing a variable before declaration? googlePlacesRef was declared on line 26 and you're using it on line 19